[DOC] Improving clarity of docs and figures (#419)
ME-ICA/tedana: docs/considerations.rst

##########################
Considerations for ME-fMRI
##########################
Multi-echo fMRI acquisition sequences and analysis methods are rapidly maturing. Someone who has access
to a multi-echo fMRI sequence should seriously consider using it. The following are a few points to 
consider when deciding whether or not to collect multi-echo data

Possible increase in TR
-----------------------

> The one difference with multi-echo is a slight time cost.

Reviewer: What about heating?

Author: I'll add some bits about possible gradient heating - maybe in the FAQ (and maybe also separated out the FAQ into multiecho specific (ie - does ME-fMRI increase heating for subjects?) and tedana specific (what is there right now).

Reviewer 2: I'd say gradient heating is an acquisition note rather than a "cost of ME", personallyl ! Maybe that's a judgement call

##########################
Considerations for ME-fMRI
##########################
Multi-echo fMRI acquisition sequences and analysis methods are rapidly maturing. Someone who has access
to a multi-echo fMRI sequence should seriously consider using it. 

The possible costs and benefits of multi-echo fMRI
==================================================
The following are a few points to consider when deciding whether or not to collect multi-echo data.

Possible increase in TR
-----------------------
The one difference with multi-echo is a slight time cost.

> For multi-echo fMRI, the shortest echo time (TE) is essentially free since it is collected in the

Reviewer 2: Should we reference Molly bright's method, here ? I think a lot of folks come to tedana thinking they only need two echos.

The second echo tends to roughly match the single-echo TE.
Additional echoes require more time.
For example, on a 3T MRI, if the T2* weighted TE is 30ms for single echo fMRI,
a multi-echo sequence may have TEs of 15.4, 29.7, and 44.0ms.
In this example, the extra 14ms of acquisition time per RF pulse is the cost of multi-echo fMRI.

One way to think about this cost is in comparison to single-echo fMRI.
If a multi-echo sequence has identical spatial resolution and acceleration as a single-echo sequence,
then a rough rule of thumb is that the multi-echo sequence will have 10% fewer slices or 10% longer TR.
Instead of compromising on slice coverage or TR, one can increase acceleration.
If one increases acceleration, it is worth doing an empirical comparison to make sure there
isn't a non-trivial loss in SNR or an increase of artifacts.

Weighted Averaging may lead to an increase in SNR 
-------------------------------------------------
Multiple studies have shown that a
weighted average of the echoes to optimize T2* weighting, sometimes called "optimally combined,"
gives a reliable, modest boost in data quality. The optimal combination of echoes can currently be

> calculated in several software packages including AFNI, fMRIPrep, and tedana. In tedana, the weighted

Reviewer 2: AFNI and fMRIPrep use tedana internally... might be worth clarifying that these are all the same method. 

There _are_ other combination methods we could reference here, though ?

average can be calculated with `t2smap`_ If no other
acquisition compromises are necessary to acquire multi-echo data, this boost is worthwhile. 

Consider the life of the dataset
--------------------------------
If other
compromises are necessary, consider the life of the data set. If data is being acquired for a discrete
study that will be acquired, analyzed, and published in a year or two, it might not be worth making
compromises to acquire multi-echo data. If a data set is expected to be used for future analyses in later
years, it is likely that more powerful approaches to multi-echo denoising will sufficiently mature and add
even more value to a data set.

> Other multi-echo denoising methods, such as MEICA, the predecessor to tedana, have shown the potential for

Reviewer 2: I'm conflicted about this paragraph. I don't know that MEICA is _more powerful_, and tedana is (hopefully, eventually) where we'll see emerging methods added. I might rephrase around these lines ?

much greater data quality improvements, as well as the ability to more accurately separate visually similar
signal vs noise, such as scanner based drifts vs slow changes in BOLD signal. These more powerful methods are
still being improved, and the algorithms are still changing. Users need to have the time and knowledge to look
at the denoising output from every run to make sure denoising worked as intended. 

> You may recover signal in areas affected by dropout

Reviewer 2: I feel like this should be next to the weighted averaging section, since they're arguably related ?

---------------------------------------------------
Typical single echo fMRI uses an echo time that is appropriate for signal across most of the brain. While this is effective
it also leads to drop out in regions with low :math:T_2^* values. This can lead to low or even no signal at all in some areas. 
If your research question could benefit from having either
improved signal characteristics in regions such as the orbitofrontal cortex, ventral temporal cortex or 
the ventral striatum them multi-echo fMRI may be beneficial. 

Consider the cost of added quality control
------------------------------------------

> If someone wants a push-button

Reviewer 2: I'd add a disclaimer here that _all data should be QC'ed_ !

Author: ALWAYS LOOK AT YOUR DATA!!!!

Author: to be clear, this is me agreeing.

way to use multi-echo data to improve data quality, that doesn't require as deep an inspection of every output,
stick with using the weighted average. The developers of tedana look forward to when tedana and other methods
have sufficiently stable algorithms, which have been validated on a wide range of data sets, so that we can
recommend the wide use of tedana.

.. _t2smap: https://tedana.readthedocs.io/en/latest/usage.html#run-t2smap

Acquisition Parameter Recommendations
=====================================
There is no empirically tested best parameter set for multi-echo acquisition.
The guidelines for optimizing parameters are similar to single-echo fMRI.
For multi-echo fMRI, the same factors that may guide priorities for single echo
fMRI sequences are also relevant.
Choose sequence parameters that meet the priorities of a study with regards to spatial resolution,
spatial coverage, sample rate, signal-to-noise ratio, signal drop-out, distortion, and artifacts.

A minimum of 3 echoes is recommended for running TE-dependent denoising.

> While there are successful studies that don’t follow this rule,

Reviewer 2: Which rule, having three echos ? Or earlier and later echos ?

Also, again, maybe Bright method ?

and use the parameters from those studies as a starting point. 
More complete recomendations
and guidelines are discussed in the `appendix`_ of Dipasquale et al, 2017.

.. _appendix: https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0173289

.. _found here: https://www.cmrr.umn.edu/multiband/
.. _this link: http://license.umn.edu/technologies/cmrr_center-for-magnetic-resonance-research-software-for-siemens-mri-scanners
.. _available here: https://www.nmr.mgh.harvard.edu/software/c2p/sms
.. _GE Collaboration Portal: https://collaborate.mr.gehealthcare.com
.. note:: 
    In order to increase the number of contrasts ("echoes") you may need to first increase the TR, shorten the 
    first TE and/or enable in-plane acceleration. 
    For typically used parameters see the `parameters and publications page`_
.. _parameters and publications page: https://tedana.readthedocs.io/en/latest/publications.html

Resources
=========

Journal articles
----------------
* | :ref:`spreadsheet of publications` catalogues papers using multi-echo fMRI,
    with information about acquisition parameters.
* | `Multi-echo acquisition`_

>   | Posse, NeuroImage 2012

Reviewer 2: ```suggestion
  |  Posse, NeuroImage 2012
```

>   | Includes an historical overview of multi-echo acquisition and research

Reviewer 2: ```suggestion
  |  Includes an historical overview of multi-echo acquisition and research
```

gives a reliable, modest boost in data quality. 
The optimal combination of echoes can currently be calculated in several software packages including AFNI, 
fMRIPrep, and tedana. In tedana, the weighted
average can be calculated with `t2smap`_ If no other
acquisition compromises are necessary to acquire multi-echo data, this boost is worthwhile. 

Consider the life of the dataset
--------------------------------
If other
compromises are necessary, consider the life of the data set. 
If data is being acquired for a discrete
study that will be acquired, analyzed, and published in a year or two, it might not be worth making
compromises to acquire multi-echo data. 
If a data set is expected to be used for future analyses in later
years, it is likely that more powerful approaches to multi-echo denoising will sufficiently mature and add
even more value to a data set.

Other multi-echo denoising methods, such as MEICA, the predecessor to tedana, have shown the potential for
much greater data quality improvements, as well as the ability to more accurately separate visually similar
signal vs noise, such as scanner based drifts vs slow changes in BOLD signal. 

> These more powerful methods are

Reviewer 2: ```suggestion
More powerful methods are
```

> still being improved, and the algorithms are still changing. 

Reviewer 2: ```suggestion
still being improved, and associated algorithms are still being actively developed. 
```

Users need to have the time and knowledge to look
at the denoising output from every run to make sure denoising worked as intended. 

You may recover signal in areas affected by dropout
---------------------------------------------------
Typical single echo fMRI uses an echo time that is appropriate for signal across most of the brain. 
While this is effective
it also leads to drop out in regions with low :math:T_2^* values. 
This can lead to low or even no signal at all in some areas. 
If your research question could benefit from having either
improved signal characteristics in regions such as the orbitofrontal cortex, ventral temporal cortex or 
the ventral striatum them multi-echo fMRI may be beneficial. 

Consider the cost of added quality control
------------------------------------------
The developers of ``tedana`` strongly support always examining data for quality concerns, whether
or not multi-echo fMRI is used. 
If someone wants a push-button
way to use multi-echo data to improve data quality, that doesn't require as deep an inspection of every output,
stick with using the weighted average. 

> The developers of tedana look forward to when tedana and other methods

Reviewer 2: This sentence feels out of place ? Since I think we should still always recommend QC. 

But I would include this sentence ! Just in the section on "the life of your dataset" ?

Author: I dropped it - because I agree strongly with your first point. Now it jsut says that mutliecho data is no exception, you should look at your data, and we produce some figures for this purpose.

have sufficiently stable algorithms, which have been validated on a wide range of data sets, so that we can
recommend the wide use of tedana.

> ``tedana`` currently allows the use of a ``--png`` flag when called, in order to produce basic diagnostic images, 

Reviewer 2: `--png` is now the default ! I'd just point people to consult them.

Author: changed the language

`see outputs`_ for more information on these outputs. 

.. _t2smap: https://tedana.readthedocs.io/en/latest/usage.html#run-t2smap
.. _see outputs: https://tedana.readthedocs.io/en/latest/outputs.html

Acquisition Parameter Recommendations
=====================================
There is no empirically tested best parameter set for multi-echo acquisition.
The guidelines for optimizing parameters are similar to single-echo fMRI.
For multi-echo fMRI, the same factors that may guide priorities for single echo
fMRI sequences are also relevant.
Choose sequence parameters that meet the priorities of a study with regards to spatial resolution,
spatial coverage, sample rate, signal-to-noise ratio, signal drop-out, distortion, and artifacts.

A minimum of 3 echoes is required for running the current implementation fo TE-dependent denoising in
``tedana``. 

> While there are successful studies that don’t follow this rule,

Reviewer 2: Can we invert this sentence ? I'm still a little confused on first read-through, personally :confused:

Author: I've removed the not follow language - and just say we need three echoes, but left in the note that dual echo is not what we are talking about here.
